integration-tests/test/test_wallets.py

```diff
@@ -0,0 +1,96 @@
+    local_contract_file_path = os.path.join('resources/wallets', contract_name)
+    node_deploy_file_path = os.path.join(node.local_deploy_dir, contract_name)
+    shutil.copyfile(local_contract_file_path, node_deploy_file_path)
+    container_contract_file_path = '{}/{}'.format(node.remote_deploy_dir, contract_name)
+    node.shell_out(
+        'sed',
+        '-i',
+        '-e', substitute_rules,
+        container_contract_file_path,
+    )
+    node.deploy(container_contract_file_path, private_key=private_key)
+    block_hash = node.propose()
+    os.remove(node_deploy_file_path)
+    return block_hash
+
+def create_genesis_vault(context: TestingContext, node: Node, rev_addr: str, init_vault_value: int) -> None:
+    log_marker = random_string(context, 10)
+    create_vault_success_pattern = re.compile("{} Genesis vault created!".format(log_marker))
+    deploy_contract_with_substitution(node, "s/%REV_ADDR/{}/g; s/%INIT_VAULT_VALUE/{}/g; s/%LOG_MARKER/{}/g".format(rev_addr, init_vault_value, log_marker), "create_genesis_vault.rho")
+    wait_for_log_match(context, node, create_vault_success_pattern)
+
+def transfer_funds(context: TestingContext, node: Node, from_rev_addr: str, to_rev_addr: str, amount: int, private_key: str) -> bool:
+    log_marker = random_string(context, 10)
+    transfer_funds_result_pattern= re.compile('"{} (Successfully|Failing) reason: (?P<reason>[a-zA-Z0-9]*)"'.format(log_marker))
+    deploy_contract_with_substitution(node, "s/%FROM/{}/g; s/%TO/{}/g; s/%AMOUNT/{}/g; s/%LOG_MARKER/{}/g".format(from_rev_addr, to_rev_addr, amount, log_marker), "transfer_funds.rho", private_key)
+    wait_for_log_match(context, node, transfer_funds_result_pattern)
+    transfer_result_match = transfer_funds_result_pattern.search(node.logs())
+    reason = transfer_result_match.group('reason')  # type:ignore
+    return reason == 'Nil'
+
+def check_balance(context: TestingContext, node: Node, rev_addr: str) -> int:
+    log_marker = random_string(context, 10)
+    check_balance_pattern = re.compile('"{} Vault (?P<rev_addr>[a-zA-Z0-9]*) balance is (?P<balance>[0-9]*)"'.format(log_marker))
+    deploy_contract_with_substitution(node, "s/%REV_ADDR/{}/g; s/%LOG_MARKER/{}/g".format(rev_addr, log_marker), "check_balance.rho")
+    wait_for_log_match(context, node, check_balance_pattern)
+    check_balance_match = check_balance_pattern.search(node.logs())
+    return int(check_balance_match.group("balance"))  # type:ignore
+
+def know_ones_rev_addr(context: TestingContext, node: Node, public_key: str) -> str:
+    log_marker = random_string(context, 10)
+    public_key_rev_addr_pattern = re.compile('"{} RevAddress for pubKey {} is (?P<rev_addr>[a-zA-Z0-9]*)"'.format(log_marker, public_key))
+    deploy_contract_with_substitution(node, "s/%PUB_KEY/{}/g; s/%LOG_MARKER/{}/g".format(public_key, log_marker), "know_ones_revaddress.rho")
```

Comment:
It would be better if `deploy_contract_with_substitution()` took a dict `{'%PUB_KEY': public_key, '%LOG_MARKER': log_marker}` and constructed the sed pattern in its body.  That makes it more structured and less of stringly-typed programming kind.

Reply:
good point